src/components/date/date.stories.tsx

```diff
@@ -0,0 +1,478 @@
+              {...{ [validationType]: "Message" }}
+              m={4}
+            />
+            <DateInput
+              label={`readOnly - ${size} - ${validationType}`}
+              value={state2}
+              onChange={setValue2}
+              validationOnLabel
+              size={size}
+              readOnly
+              {...{ [validationType]: "Message" }}
+              m={4}
+            />
+          </div>
+        ))
+      )}
+    </CarbonProvider>
+  );
+};
+
+export const ValidationsBoolean: ComponentStory<typeof DateInput> = () => {
+  const [state1, setState1] = useState("01/10/2016");
+  const setValue1 = (ev: DateChangeEvent) => {
+    setState1(ev.target.value.formattedValue);
+  };
+  const [state2, setState2] = useState("01/10/2016");
+  const setValue2 = (ev: DateChangeEvent) => {
+    setState2(ev.target.value.formattedValue);
+  };
+  return (
+    <>
+      {["error", "warning", "info"].map((validationType) => (
+        <div key={`${validationType}-boolean-component`}>
+          <DateInput
+            label="Date"
+            value={state1}
+            onChange={setValue1}
+            {...{ [validationType]: true }}
+          />
+          <DateInput
+            label="Date"
+            value={state2}
+            onChange={setValue2}
+            readOnly
+            {...{ [validationType]: true }}
+          />
+        </div>
+      ))}
+    </>
+  );
+};
+
+export const ValidationsExampleImplementation: ComponentStory<
+  typeof DateInput
+> = () => {
+  const [state, setState] = useState("05/04/2022");
+  const [validationState, setValidationState] = useState("");
+  const handleChange = (ev: DateChangeEvent) => {
+    setState(ev.target.value.formattedValue);
+  };
+  const handleBlur = (ev: DateChangeEvent) => {
+    if (!ev.target.value.rawValue) {
+      setValidationState("Error Invalid Date");
+    } else if (new Date(ev.target.value.rawValue).getFullYear() <= 2020) {
+      setValidationState("Warning Date before 2020");
+    }
+  };
+  return (
+    <div>
+      <DateInput
+        label="Date"
+        value={state}
+        onChange={handleChange}
+        onBlur={handleBlur}
+        error={validationState.includes("Error") ? validationState : undefined}
+        warning={
+          validationState.includes("Warning") ? validationState : undefined
+        }
+      />
+    </div>
+  );
+};
+
+ValidationsExampleImplementation.parameters = {
+  chromatic: { disableSnapshot: true },
+};
+
+export const LocaleOverrideExampleImplementation: ComponentStory<
+  typeof DateInput
+> = () => {
+  const [state, setState] = useState("2022-04-05");
+  const handleChange = (ev: DateChangeEvent) => {
+    setState(ev.target.value.formattedValue);
+  };
+  const [state2, setState2] = useState("2022-04-05");
+  const handleChange2 = (ev: DateChangeEvent) => {
+    setState2(ev.target.value.formattedValue);
+  };
+  return (
+    <div style={{ display: "flex", justifyContent: "space-around" }}>
```

Comment:
suggestion: Is it worth changing this to `Box`? I believe @Parsium has a PR pending to swap out all instances of HTML elements in out stories.

Reply:
Yeah that's why I've left it as wasn't sure what the state of his work was in etc, happy to update if James gives the green light

Reply:
Wonderful. Either which way is good for me 😄